### ISSUE TITLE
Full results pages for large competitions need to be split into multiple pages

### ISSUE THREAD

**Issue by ToastyKen** [thewca/worldcubeassociation.org#2122]
Example, the Worlds 2017 full results page:
https://www.worldcubeassociation.org/competitions/WC2017/results/all

On mobile, it pretty much kills my phone.  Even on desktop, it's extremely slow.

Proposed solution is to have a mode where each round of each event is split into a separate page, ideally with an easy table of contents navigation box on the left or something.  Basically, do it like CubeComps, which works fine.

**Comment by jfly** [comment]
Really? For me on both mobile and desktop, I start seeing results almost immediately, and I can even scroll and click around while the rest of the page loads. I think it's a pretty decent experience in both environments.

It does take a while for the page to fully load, but I don't think that's a big problem.

**Comment by ToastyKen** [comment]
Well, my phone has been slow in general lately, but I've definitely experienced serious lag before, especially when I wasn't on WiFi.  It does seem relatively responsive on desktop right now.

I've edited the original description to focus on the main problem, which is that it's difficult to find a specific round, because rounds are not indexed.

For instance, go to that page and try to find Rubik's Cube Round 3.  My experience with this in the modified original post.

**Comment by ToastyKen** [comment]
Oh here we go. I turned of WiFi, tried to load that page, and, after loading some of the page, it literally crashed Chrome on Android:

https://photos.app.goo.gl/6VLOdU2c8nsmJeUH2

**Comment by ToastyKen** [comment]
That page is over 4 MB of HTML!

**Comment by jfly** [comment]
That's pretty lame that it crashes your android phone =(

> That page is over 4 MB of HTML!

Sure, but it's being transferred gzipped, and we get huge savings that way.

You can see here that it's only 467 KB of download to load the page:

![image](https://user-images.githubusercontent.com/277474/32311958-6d367630-bf58-11e7-8618-55fc2e48c935.png)

I did this load with nothing cached, and you can see that the total amount of bandwidth to load the page is "only" 1.4 MB.

**Comment by ToastyKen** [comment]
So that's a matter of degree, but what about the main issue, which is that there are no links to individual rounds, and scrolling to find them is onerous for large comps?

I think supporting a broader set of links on the left would be an easy solution to this (compared to putting each on its own page, which I imagine would be harder).

**Comment by ToastyKen** [comment]
Btw, I don't think it's the file size that's the issue with the 4 MB so much as the RAM and CPU load once you render that into a page, which is probably what killed Chrome on my phone. :\

**Comment by ToastyKen** [comment]
Anyway, again, back to the main issue: Have you tried scrolling down the page? It's ridiculously long. Surely we can agree the usability could be improved?

**Comment by jfly** [comment]
> I think supporting a broader set of links on the left would be an easy solution to this (compared to putting each on its own page, which I imagine would be harder).

Yes! I have also found this quite annoying. Another idea might be to provide "next/previous" round links when you're looking at a particular round, which would let you get around faster.

> Btw, I don't think it's the transfer time that's the issue with the 4 MB so much as the RAM and CPU load once you render that into a page, which is probably what killed Chrome on my phone. :\

Yeah, I think the only fix for that is to split this up into multiple pages, which I'm not opposed to, but maybe should be a different project.

**Comment by BiscuitCubed** [comment]
Another idea might be to collapse the actual results under their round heading. That way you can quickly navigate to the event/round you want to see, click on it, and then it expands to show all results from that round. Shouldn't be too hard to do either.

If you wanted to get really fancy, you could only do that for a competition with x results, and/or y events.  Honestly, I really dislike the way it is now, and this makes it feel better organized and less cluttered IMO.

**Comment by AlbertoPdRF** [comment]
With the work of @timreyn on #4517, can we consider this issue fixed?
